### PR TITLE
Add tsdb metrics builtin component template

### PR DESCRIPTION
**Pull request by martijnvg**
Fleet is currently hard coded to set `index.codec` to `best_compression` (deflate compression). This is good for most data streams, except for data streams were tsdb is enabled. Ideally Fleet doesn't need to set this setting at all and Elasticsearch's default would be good. But unfortunately this isn't the case. It default to `default` (lz4 - optimised for speed), which in would mean much higher disk space usage. Ideally the default would be `default` when synthetic source is enabled and otherwise `best_compression.` Changing this now, would mean a breaking change.

Instead Fleet like to depend on Elasticsearch's internal component templates. To at least abstract some of the internal details away. The `metrics-settings` is ok for non tsdb, but there is no component template for tsdb metrics. This PR adds this.

Relates to elastic/kibana#160288

### CODE REVIEW

**Review by felixbarny — APPROVED** [review]
You'll probably need to update 

https://github.com/elastic/elasticsearch/blob/5ebfddad0e073a3c552fa29e05401f24e88ab889/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java#L1842

And add `metrics-tsdb-settings`.

Otherwise LGTM

**Comment by ruflin** [comment]
@eyalkoren New built in template here. Can you check that it aligns with the naming convention?

**Comment by eyalkoren** [comment]
> @eyalkoren New built in template here. Can you check that it aligns with the naming convention?

Is there a naming convention already? 🤔

**Comment by ruflin** [comment]
Lets align it with the hopefully soon to be reached conclusion in https://github.com/elastic/elasticsearch/issues/96267

**Review by andreidan — APPROVED** [review]
LGTM, thanks Martijn

File: x-pack/plugin/core/template-resources/src/main/resources/metrics-tsdb-settings.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "template": {
+    "settings": {
+      "index": {
+        "lifecycle": {
```
[the remark this answers is not in the document]
Comment (martijnvg) [reply x-pack/plugin/core/template-resources/src/main/resources/metrics-tsdb-settings.json:5]: iirc this will be used with data streams.

**Pull request merged**